docs/resolver-template/cmd/demoresolver/main.go

```diff
@@ -51,16 +51,16 @@ func (r *resolver) GetSelector(context.Context) map[string]string {
 	}
 }
 
-// ValidateParams ensures parameters from a request are as expected.
-func (r *resolver) ValidateParams(ctx context.Context, params []pipelinev1.Param) error {
-	if len(params) > 0 {
+// Validate ensures resolution spec from a request is as expected.
+func (r *resolver) Validate(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error {
```

Comment:
No, both of those are valid. Billy suggested having a separate package to showcase the new interface and leave the old one as is so I will do that to avoid confusion.